core/text/table/table.odin

```diff
@@ -152,22 +164,116 @@ header :: proc(tbl: ^Table, values: ..any, loc := #caller_location) {
 	tbl.dirty = true
 }
 
-row :: proc(tbl: ^Table, values: ..any, loc := #caller_location) {
+aligned_header_of_values :: proc(tbl: ^Table, alignment: Cell_Alignment, values: ..any, loc := #caller_location) {
+	if (tbl.has_header_row && tbl.nr_rows != 1) || (!tbl.has_header_row && tbl.nr_rows != 0) {
+		panic("Cannot add headers after rows have been added", loc)
+	}
+
+	if tbl.nr_rows == 0 {
+		tbl.nr_rows += 1
+		tbl.has_header_row = true
+	}
+
+	col := tbl.nr_cols
+	tbl.nr_cols += len(values)
+	for val in values {
+		set_cell_value_and_alignment(tbl, header_row(tbl), col, val, alignment, loc)
+		col += 1
+	}
+
+	tbl.dirty = true
+}
+
+header_of_aligned_values :: proc(tbl: ^Table, aligned_values: []Aligned_Value, loc := #caller_location) {
+	if (tbl.has_header_row && tbl.nr_rows != 1) || (!tbl.has_header_row && tbl.nr_rows != 0) {
+		panic("Cannot add headers after rows have been added", loc)
+	}
+
+	if tbl.nr_rows == 0 {
+		tbl.nr_rows += 1
+		tbl.has_header_row = true
+	}
+
+	col := tbl.nr_cols
+	tbl.nr_cols += len(aligned_values)
+	for av in aligned_values {
+		set_cell_value_and_alignment(tbl, header_row(tbl), col, av.value, av.alignment, loc)
+		col += 1
+	}
+
+	tbl.dirty = true
+}
+
+row :: row_of_values
+row_of_values :: proc(tbl: ^Table, values: ..any, loc := #caller_location) {
+	if tbl.nr_cols == 0 {
+		if len(values) == 0 {
+			panic("Cannot create empty row unless the number of columns is known in advance")
+		} else {
+			tbl.nr_cols = len(values)
+		}
+	}
+
+	tbl.nr_rows += 1
+
+	for col in 0 ..< tbl.nr_cols {
+		val := values[col] if col < len(values) else nil
+		set_cell_value(tbl, last_row(tbl), col, val, loc)
+	}
+
+	tbl.dirty = true
+}
+
+aligned_row_of_values :: proc(tbl: ^Table, alignment: Cell_Alignment, values: ..any, loc := #caller_location) {
 	if tbl.nr_cols == 0 {
 		if len(values) == 0 {
 			panic("Cannot create empty row unless the number of columns is known in advance")
 		} else {
 			tbl.nr_cols = len(values)
 		}
 	}
+
 	tbl.nr_rows += 1
-	for col in 0..<tbl.nr_cols {
+
+	for col in 0 ..< tbl.nr_cols {
```

Comment:
`a..<b` (without surrounding space) is the preferred Odin style.

Reply:
I find this style `a ..< b` to be more readable and tend to use it. I did a search to confirm your statement. At the very least, it's more common (~450 instances of `a..<b` versus ~50 of `a ..< b`). I can change it if it's an issue.